src/utils/DatePick.tsx

```diff
@@ -0,0 +1,71 @@
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import { ko } from 'date-fns/locale'
+import inputCalendar from '@/images/inputCalendar.svg'
+import { forwardRef, useState } from 'react'
+import { format } from 'date-fns'
+
+interface DatePickProps {
+	selectedDate: Date
+	minDate?: Date
+	maxDate?: Date
+	textSize?: string
+	handleSelect: (date: Date) => void
+}
+
+interface DateInputProps {
+	onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+export const DatePick = ({
+	selectedDate,
+	minDate = new Date(0, 0),
+	maxDate = new Date(),
+	handleSelect,
+	textSize = 'text-md',
+}: DatePickProps) => {
+	const [open, setOpen] = useState(false)
+
+	const DateInput = forwardRef<HTMLButtonElement, DateInputProps>(
+		({ onClick }, ref) => {
+			const formattedDate = selectedDate
+				? format(selectedDate, 'yyyy-MM-dd')
+				: ''
+
+			return (
+				<button
+					className='example-custom-input hover:bg-gray-200 hover:duration-300'
+					onClick={onClick}
+					ref={ref}
+				>
+					{formattedDate}
+				</button>
+			)
+		},
+	)
+
+	return (
+		<DatePicker
+			icon={<img src={inputCalendar} />}
+			className={`${textSize}`}
+			// 기본 Input창의 패딩이 너무 큰데, 이걸 수정할 수가 없어서 커스텀 인풋을 만듬
+			customInput={<DateInput onClick={() => setOpen(true)} />}
+			locale={ko}
+			dateFormat='yyyy-MM-dd'
+			//state를 넘길지 말지 고민 많이 했는데 넘기는게 맞는듯, 이거 아니면 오히려 복잡한데 결국 상태를 받아야함
+			selected={selectedDate}
+			open={open}
+			minDate={minDate}
+			maxDate={maxDate}
+			onInputClick={() => setOpen(true)}
+			onClickOutside={() => setOpen(false)}
+			onSelect={(date) => {
+				setOpen(false)
+				if (!date) return
+				handleSelect(date)
```

Comment:
`handleSelect`에 정의해주신 타입을 보니 항상 `Date` 타입만을 받고 있습니다. `Date | null`, `Date | undefined`의 형식이 아니기 때문에 여기서의 `date`는 항상 `Date` 타입이 됩니다. 불필요한 예외처리인 것 같습니다.

만약 사용자가 날짜를 선택하지 않은 경우에는 `onSelect` 자체가 호출되지 않습니다. 날짜가 선택된 경우에만 `onSelect`가 호출되고, 이때 전달되는 `date`는 항상 유효한 `Date` 객체입니다.
```suggestion
				setOpen(false)
				handleSelect(date)
```

Reply:
onSelect가 Date|null을 인자로 갖는 콜백 함수가 들어가야하기 때문에, 그에 맞게끔 handleSelect의 타입을 수정하고 단순 set이 아닌 에러를 처리하는 함수를 받게끔 수정했습니다.

저도 setState를 직접 넘기는건 위험하다고 생각했고 수정할 방안을 찾고 있었습니다.

<img width="742" alt="image" src="https://github.com/user-attachments/assets/93b0ab22-6980-48fc-b329-f1ceb3759e68" />
이런 식으로 context에서 setState를 처리한 후 넘기게끔 수정했습니다!

Reply:
네 이렇게 핸들러를 분리해서 그쪽에서 props를 Date | null 로 받는 형식이 좋을 것 같습니다. set 함수의 은닉도 되구요.